DataGrid/Extension/Symfony/EventSubscriber/FormSubscriber.php

```diff
@@ -0,0 +1,173 @@
+use Symfony\Component\HttpFoundation\Request;
+
+class FormSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var FormFactoryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @var bool
+     */
+    private $csrfTokenEnabled;
+
+    /**
+     * @var FormInterface[]
+     */
+    private $forms;
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            DataGridEvents::POST_SET_DATA => 'buildForms',
+        ];
+    }
+
+    public function __construct(FormFactoryInterface $formFactory, bool $csrfTokenEnabled = true)
+    {
+        $this->formFactory = $formFactory;
+        $this->csrfTokenEnabled = $csrfTokenEnabled;
+        $this->forms = [];
+    }
+
+    public function buildForms(DataGridEvent $event): void
+    {
+        $dataGrid = $event->getDataGrid();
+        $data = $event->getData();
+
+        $entryType = $this->isSymfony3() ? RowType::class : 'row';
+
+        if ($this->csrfTokenEnabled) {
+            $formBuilderOptions['csrf_protection'] = false;
+        }
+
+        $formBuilder = $this->formFactory
+            ->createNamedBuilder($dataGrid->getName(), $this->getFormTypeName(), $data, $formBuilderOptions);
+
+        foreach ($data as $index => $row) {
+            $formBuilder->add($index, $entryType, [
+                'data_class' => get_class($row),
+                'property_path' => sprintf('[%s]', $index),
+            ]);
+            $rowFormBuilder = $formBuilder->get($index);
+
+            foreach ($dataGrid->getColumns() as $column) {
+                if (!$column->hasOption('editable') || !$column->getOption('editable')) {
+                    continue;
+                }
+
+                $rowFormBuilder->add($column->getName(), $this->getFormTypeName(), ['inherit_data' => true]);
+                $cellFormBuilder = $rowFormBuilder->get($column->getName());
+
+                $columnTypeExtensions = $dataGrid->getFactory()->getColumnTypeExtensions($column->getType());
+                foreach ($columnTypeExtensions as $columnTypeExtension) {
+                    if (!$columnTypeExtension instanceof CellFormBuilder) {
+                        continue;
+                    }
+
+                    $columnTypeExtension->buildCellForm($cellFormBuilder, $column);
+                }
+            }
+        }
+
+        $this->forms[$dataGrid->getName()] = $formBuilder->getForm();
+    }
+
+    public function handleDataGridForm(DataGrid $dataGrid, Request $request): bool
+    {
+        if (!array_key_exists($dataGrid->getName(), $this->forms)) {
+            throw new \RuntimeException(sprintf(
+                'DataGrid "%s" does not have associated form builder',
+                $dataGrid->getName()
+            ));
+        }
+
+        $dataGridForm = $this->forms[$dataGrid->getName()];
+        $submittedData = $request->request->get($dataGrid->getName());
+        if (null === $submittedData) {
```

Comment:
It may be better to simply call `$request->request->has($dataGrid->getName());` and only read the data if it does. Also, why manually checking for submitted data and removing it from the datagrid?